samples/snippets/fsharp/VS_Snippets_CLR_System/System.AppContext.Class/fs/Example6.fs

```diff
@@ -0,0 +1,30 @@
+module Example6
+
+// <Snippet6>
+open System
+open System.Reflection
+
+[<assembly: AssemblyVersion("2.0.0.0")>]
+do ()
+
+module StringLibrary =
+    let substringStartsAt (fullString: string) (substr: string) =
+        fullString.IndexOf(substr, StringComparison.CurrentCulture)
+
+// </Snippet6>
+
+// <Snippet7>
+let value = "The archaeologist"
+let substring = "archæ"
+
+let position =
+    StringLibrary.substringStartsAt value substring
```

Comment:
Why define "StringLibrary"?  Just call "IndexOf directly?

Reply:
The article demonstrates how a library may use Appcontext switches to vary calling behavior. Snippets 4 and 6 demonstrate different iterations of the same library with different behavior. Snippet 8 shows how to choose behavior based on a switch. While the calling IndexOf directly would work, it wouldn't fit with the article.